include/ur_api.h

```diff
@@ -219,6 +219,7 @@ typedef enum ur_result_t {
     /// @endcond
 
 } ur_result_t;
+#define UR_RESULT_MAX_VALUE UR_RESULT_ERROR_UNKNOWN
```

Comment:
This specific enum has `UR_RESULT_ERROR_UNKNOWN = 0x7ffffffe,` (which is much larger than the actual last value of `UR_RESULT_ERROR_ADAPTER_SPECIFIC = 67,`. This means that `if (val > UR_RESULT_ERROR_UNKNOWN) {...}` won't actually catch all invalid cases.

Not sure if this is much of a problem for `ur_result_t`, but other enums might have the same issue. Perhaps it would be better to use 0 (or just generally, the beginning of the enum) for special values like `UR_RESULT_ERROR_UNKNOWN` ?

Reply:
tbh I want this mostly just for `flags_t` enumerations - I extended it a bit so that it can support an arbitrary enum, but as you say doesn't work that well if the enum has gaps.